### PR TITLE
Address (some) TR comments for chapter 7

### PR DESCRIPTION
**Pull request by jni**
Figure captions and alignment for the multi-method plot I'm punting on.

### CODE REVIEW

File: markdown/ch7.markdown [hunk 1]
```diff
@@ -342,12 +342,13 @@ def align(reference, target, cost=cost_mse):
         res = optimize.minimize(cost, p, args=(ref, tgt), method='Powell')
         p = res.x
 
-        print('Pyramid level %i' % n)
-        print('Angle:', np.rad2deg(res.x[0]))
-        print('Offset:', res.x[1:] * 2 ** n)
-        print('Cost function:', res.fun)
-        print('')
+        # print current level, overwriting each time (like a progress bar)
+        print('Level: %i, Angle: %.2f, Offset: (%.1f, %.1f), Cost: %.2f' %
```
Comment (stefanv) [inline markdown/ch7.markdown]: My immediate thought was: formatting!  But looking at it now, it actually looks worse :)

```
state = {'level': n,
         'angle': np.rad2deg(res.x[0]),
         'offset': (res.x[1] ** 2**n, res.x[2] * 2**n),
         'cost': res.fun}

print('Level: {level:i}, Angle: {angle:.2f}, '
      'Offset: ({offset[0]:.1f}, {offset[1]:.1f}), '
      'Cost: {cost:%.2f'.format(**state))

# or

print('Level: {0:i}, Angle: {angle:.2f}, '
      'Offset: ({2:.1f}, {3:.1f}), Cost: {4:.2f}'.format(
          n, np.rad2deg(res.x[0]),
          res.x[1] ** 2**n, res.x[2] ** 2**n,
          res.fun))
```

Comment (jni) [inline markdown/ch7.markdown]: Yes, formatting is terrible, I don't know who thought it was an improvement. f-strings, on the other hand, might be nice, but then we are 3.6-only:

```python
print(f'Level: {n}, Angle: {np.rad2deg(res.x[0]):.3}, '
      f'Offset: ({res.x[1] * 2**n:.3}, {res.x[2] * 2**n:.3}), '
      f'Cost: {res.fun:.3}', end='\r')
```

Comment (jni) [inline markdown/ch7.markdown]: (Which I'm ok with, btw! ;)

**Comment by stefanv** [comment]
Heh, me too ;)

**Comment by jni** [comment]
@stefanv so do you want me to update it, or forget it for now? If so feel free to merge...! =)

File: markdown/ch7.markdown (hunk 1 above)
Comment (hdashnow) [inline markdown/ch7.markdown]: I'm okay with Python 3.6

Comment (jni) [inline markdown/ch7.markdown]: And what do you think of the f-string's readability vs %?

Comment (hdashnow) [inline markdown/ch7.markdown]: I think the f-strings are a huge improvement. I've never used % in my own code, only .format(). I feel lucky that I arrived on the Python scene just late enough to have never been required to use %.

Comment (jni) [inline markdown/ch7.markdown]: Ok, hold please. Updating chapter to use f-strings throughout. Yay!

**Comment by jni** [comment]
@hdashnow this is ready imho — feel free to merge when/if the build completes! =)

File: markdown/ch7.markdown [hunk 2]
```diff
@@ -342,12 +342,12 @@ def align(reference, target, cost=cost_mse):
...
 
-        print('Pyramid level %i' % n)
-        print('Angle:', np.rad2deg(res.x[0]))
-        print('Offset:', res.x[1:] * 2 ** n)
-        print('Cost function:', res.fun)
-        print('')
+        # print current level, overwriting each time (like a progress bar)
+        print(f'Level: {n}, Angle: {np.rad2deg(res.x[0]) :.3}, '
+              f'Offset: ({res.x[1] * 2**n :.3}, {res.x[2] * 2**n :.3}), '
+              f'Cost: {res.fun :.3}', end='\r')
```
Comment (stefanv) [inline markdown/ch7.markdown]: Space?

Comment (jni) [inline markdown/ch7.markdown]: @stefanv where? You mean before the colon? I thought it best to separate the formatting `:.3` from the expression being evaluated.

Comment (stefanv) [inline markdown/ch7.markdown]: Ah, okay, I didn't even know that was an option.  I'm not opposed.

Comment (hdashnow) [inline markdown/ch7.markdown]: More changes required @stefanv, or shall I merge?

**Comment by hdashnow** [comment]
Does Wercker need to be updated to Python 3.6 before this will build?

**Comment by jni** [comment]
Possibly... @stefanv I don't see a place to specify Python version for Wercker...?

**Comment by jni** [comment]
Ah, it's our docker box... @stefanv how do we update this? I note that it apt-get installs Python... Has 3.6 been propagated yet?

**Comment by stefanv** [comment]
I'm on it.

**Pull request merged**